### PR TITLE
feat(user-input-mvp): user input

### PR DESCRIPTION
**Pull request by jaycosaur**

### CODE REVIEW

File: functions/src/userInput/notifyUserInputRequest.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { publishToChannel } from '@shared/channels';
+import { UserInput } from '@aimee-blue/ab-shared';
+
+const defaultDeps = {
+  publishToChannel,
+};
+
+interface INotifyUserInputRequestParams extends UserInput.IRequestPayload {
+  channel?: string;
```
Comment (zaripych) [inline functions/src/userInput/notifyUserInputRequest.ts]: I wonder when the channel is not same as `conversationId`? I think even for testing purposes we can just set `conversationId` to whatever we want ...

File: functions/src/userInput/public/userInputRequestAsHil.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const uid = 'auth' in context && context.auth.uid;
+  if (!uid) {
+    throw new Error('No uid');
+  }
+
+  const params = validate(rawParams, UserInput.requestParamsSchema, 'public');
+
+  await deps.notifyUserInputRequest({ ...params });
+
+  return ResponseStatus.success;
```
Comment (zaripych) [inline functions/src/userInput/public/userInputRequestAsHil.ts]: I saw you always returning this, but I'm not sure why. Which party needs it? The status can be evident from the `status` of `HTTP` request ... which is what almost every time we rely on. Is it just convention for the sake of convention?

File: functions/src/userInput/notifyUserInputRequest.ts (hunk 1 above)
Comment (zaripych) [reply functions/src/userInput/notifyUserInputRequest.ts]: I saw lots of other entities have `channel` as well ... when not really required.

Comment (zaripych) [reply functions/src/userInput/notifyUserInputRequest.ts]: No need to change anything for now, we need to think about when we need `channel` and when we don't need it.

File: functions/src/userInput/public/userInputRequestAsHil.ts (hunk 2 above)
Comment (jaycosaur) [reply functions/src/userInput/public/userInputRequestAsHil.ts]: You betcha! We can just return null after all. This is used everywhere though. So another PR to fix them on all onCall functions.

**Pull request merged**